umbraco-cms/tutorials/creating-an-xml-site-map.md

```diff
@@ -33,171 +34,194 @@ See [Sitemaps XML format](https://www.sitemaps.org/protocol.html) for the XML sc
+{% endhint %}
 
-#### Looping through the rest of the site
+Visit the URL of your sitemap page (`http://yoursite.com/sitemap`) to render a single sitemap entry for the homepage.
 
-So now we need to find the pages created beneath the homepage, and see if they should be added to the sitemap, and then in turn look at the pages beneath those etc, until the entire content tree is traversed.
+![Example of Sitemap](images/sitemap.png)
 
-We can use **IPublishedContent**'s _.Children_ method to return all the pages directly beneath a particular page eg:
+### Looping through the rest of the site
 
-```
-IEnumerable<IPublishedContent> sitePages = siteHomePage.Children();
-```
+We need to go through each page created beneath the homepage to see if they should be added to the sitemap.
 
-So we need to loop through each of these 'child' pages, and write out their sitemap markup using our helper, and then in turn loop through their children (grandchildren?) etc and so on... (great-great-grandchildren...)
+We can use the `.Children` method from `IPublishedContent` to return all the pages directly beneath a particular page.
 
-```csharp
-foreach (var page in sitePages){
-    @RenderSiteMapUrlEntry(page)
-    if (page.Children.Any()){
-        var childPages = page.Children();
-        foreach (var childPage in childPages){
-            @RenderSiteMapUrlEntry(childPage)
-            if (childPage.Children.Any()){
-                var grandChildPages = childPage.Children();
-                foreach (var grandChildPage in grandChildPages){
-                    // ... on forever how do we stop?
+1. Add `IEnumerable<IPublishedContent> sitePages = siteHomePage.Children();` within the first set of curly brackets in the template.
+2. Loop through each of these 'child' pages, and write out their sitemap markup using the helper:
+
+    ```csharp
+    foreach (var page in sitePages){
+        @RenderSiteMapUrlEntry(page)
+        if (page.Children.Any()){
+            var childPages = page.Children();
+            foreach (var childPage in childPages){
+                @RenderSiteMapUrlEntry(childPage)
+                if (childPage.Children.Any()){
+                    var grandChildPages = childPage.Children();
+                    foreach (var grandChildPage in grandChildPages){
+                        // ... can go on forever. How do we stop?
+                    }
                 }
             }
         }
     }
-}
-```
+    ```
 
-So hopefully you can see the problem here, how deep do we go? How do we handle the repetition forever...
+Do you see the potential problem here? How deep do we go and how do we handle the repetition?
 
-... well we can use recursion - we can create a further razor helper that 'calls itself' \[insert inception reference here]...
+One option is to use recursion. We can create a Razor helper that 'calls itself'.
 
-#### Recursive Helper
+{% hint style="warning" %}
```

Comment:
I understand we are trying to show that this is one way of doing it... but I'm not sure if we really want users to do it and remove the code. There is a possibility they might just skip through paras/info and only look at the code. Maybe just skip that part and show what is supposed to be done since it's a tutorial and one might follow all the steps mentioned here 🙈